src/utilities/transition/BaseTransition.js

```diff
@@ -0,0 +1,250 @@
+'use strict';
+
+// Required modules.
+var EventObserver = require( '../../mixins/Events.js' );
+var fnBind = require( '../function-bind' ).bind;
+
+/**
+ * BaseTransition
+ * @class
+ *
+ * @classdesc Initializes new BaseTransition behavior.
+ *   This shouldn't be used directly, but instead should be
+ *   the base class used through composition by a specific transition.
+ *
+ * @param {HTMLNode} element
+ *   DOM element to apply transition to.
+ * @param {Object} classes
+ *   The classes to apply to this transition.
+ * @returns {BaseTransition} An instance.
+ */
+function BaseTransition( element, classes ) { // eslint-disable-line max-statements, no-inline-comments, max-len
+  var _classes = classes;
+  var _dom;
+
+  var _lastClass;
+  var _transitionEndEvent;
+  var _transitionCompleteBinded;
+  var _addEventListenerBinded;
+  var _isAnimating = false;
+  var _isFlushed = false;
+
+  /**
+   * @returns {BaseTransition} An instance.
+   */
+  function init() {
+    _transitionCompleteBinded = fnBind( _transitionComplete, this );
+    _addEventListenerBinded = fnBind( _addEventListener, this );
+    setElement( element );
+
+    return this;
+  }
+
+  /**
+   * Set the HTML element target of this transition.
+   * @param {HTMLNode} elem - The target of the transition.
+   */
+  function setElement( elem ) {
```

Comment:
I named it `elem` here because this is a method of a constructor function that has `element` as a parameter of the constructor enclosing this method. So having it here means it's duplicating the name in the outer scope. Maybe name it `element` here and `initialElement` on https://github.com/cfpb/AtomicComponent/pull/1/files#diff-8e86f22986223a76cfacd9fd2191f99dR21